src/main/java/agh/ics/oop/Animal.java

```diff
@@ -0,0 +1,46 @@
+package agh.ics.oop;
+
+public class Animal {
+
+    private MapDirection orientation;
+    private Vector2d position;
+
+    Vector2d lowerLeft = new Vector2d(0, 0);
+    Vector2d upperRight = new Vector2d(4, 4);
+
+
+    public Animal() {
+        orientation = MapDirection.NORTH;
+        position = new Vector2d(2, 2);
+    }
+    public String toString() {
+        return "Zwierze znajduje się w punkcie: " + this.position + ". Jest skierowany na " + this.orientation;
+    }
+
+    public boolean isAt(Vector2d position) {
+        return this.position.equals(position);
+    }
+
+    public void setPosition(Vector2d position){
```

Comment:
niby ma to zabezpiecznie przed rozspójnieniem, ale nie dawałbym innego publicznego sposobu na modyfikację pozycji niż metoda `move()`. Jako pomocnicza metoda może być, ale wtedy prywatna.